frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-pool.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { NS, CLUSTER_STATUS } from '../utils/consts';
+import {
+  cancelAction,
+  confirmAction,
+  createStorageClassButton,
+  emptyStateBody,
+  storagePoolDropdown,
+  storagePool,
+} from '../views/multiple-pool';
+
+describe('Test ceph pool creation', () => {
+  
+  // Ceph status var
+  let cephStatus:any;
+
+  // Pool var
+  const poolName: string = 'example.com';
+  const replicaCount: string = '2';
+  const provisioner: string = 'openshift-storage.rbd.csi.ceph.com';
+
+  before(() => {
+    cy.login();
+    cy.visit('/');
+    // cy.install();
+
+    // Check ceph cluster status
+    cy.exec(`oc get cephCluster -n ${NS} -o json`).then((res) => {
+      const cephValue = JSON.parse(res.stdout.toString());
+      cephStatus = cephValue.items[0];
+    });
+  });
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.clickNavLink(['Storage', 'Storage Classes']);
+
+    // View create storage class form
+    cy.byTestID(createStorageClassButton).click();
+  });
+
+  after(() => {
+    storagePool.delete(poolName);
+    cy.logout();
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  it('Check for a new pool creation', () => {
+    if (cephStatus?.status?.phase === CLUSTER_STATUS.READY) {
+      cy.log('Creating a new pool when cluster is in ready state');
+      storagePool.prepareStorageClassForm(provisioner);
+      storagePool.create(poolName, replicaCount);
+  
+      // Validations
+      storagePool.validate(emptyStateBody, storagePool.getPoolMessage(poolName, "POOL_START"));
+      storagePool.validate(emptyStateBody, storagePool.getPoolMessage(poolName, "POOL_CREATED"));
+      // Close pool creation form
+      cy.byTestID(confirmAction).click();
+      // Should display a new pool name in dropdown
+      cy.byTestID(storagePoolDropdown).click();
+      storagePool.validate(poolName, poolName);
+    } else if (cephStatus?.status?.phase !== CLUSTER_STATUS.READY) {
+        // This block will execute only if cluster is not in ready state
+        cy.log('Trying to create a new pool when cluster is not in ready state');
+        storagePool.prepareStorageClassForm(provisioner);
+
+        // Validations
+        storagePool.validate(emptyStateBody, storagePool.getPoolMessage(poolName, "PROGRESS"));
+        // Close pool creation form
+        cy.byTestID(cancelAction).click();
+    };
+  });
+
+  it('Should successfully created in backend', () => {
+    console.log('Verifying a pool created in backend or not');
+    storagePool.checkBackend(poolName);
+  });
+
+  it('Should throw an error if duplicate pool is created', () => {
+    cy.log('Try to creating a new pool with already existing name');
+    storagePool.prepareStorageClassForm(provisioner);
+    storagePool.create(poolName, replicaCount);
+
+    // Validations
+    storagePool.validate(emptyStateBody, storagePool.getPoolMessage(poolName, "POOL_DUPLICATED"));
+    // Close pool creation form
+    cy.byTestID(confirmAction).click();
+  });
```

Comment:
```suggestion
  it('Should throw an error if duplicate pool is created', () => {
    cy.log('Try to creating a new pool with already existing name');
    storagePool.prepareStorageClassForm(provisioner);
    storagePool.create(poolName, replicaCount);

    // Validations
    storagePool.validate(emptyStateBody, storagePool.getPoolMessage(poolName, "POOL_DUPLICATED"));
    // Close pool creation form
    cy.byTestID(confirmAction).click();
  });
```
This too should be part of the same `it` block. As for this to work we need the previous `it` block to run.